### ISSUE TITLE
Feature Request: Save Protection/Validation

### ISSUE THREAD

**Issue by 0xABADCAFE** [AndyLoft/AB3D2_LevelED#28]
**TL;DR** - prevent accidental damage caused by saving data to the wrong level slot.

After loading an existing level, keep track of which level it was, so that:

1) Attempting to call "build" or "build full" triggers a warning if choosing a different slot. The user has the chance then to cancel, or proceed with overwriting a different level slot.

2) Attempting to call "save", "build clips file" or "link level" for a different slot will display an error dialogue and not be allowed as it would result in a slot containing data for two different levels, which suffice to say, would be a disaster.

When the level is saved, whatever slot was ultimately used becomes the new slot to keep track of.

When starting a brand new level, having not loaded one:

1) Attempting to call "build clips file" or "link level" will ensure the level is saved first so that there's no risk of partial data (it may already do this TBH).

**Comment by AndyLoft** [comment]
started on this
https://github.com/AndyLoft/AB3D2_LevelED/tree/lev_save_opts

**Comment by AndyLoft** [comment]
done.
